src/cascadia/WindowsTerminal/NonClientIslandWindow.cpp

```diff
@@ -38,162 +34,102 @@ NonClientIslandWindow::~NonClientIslandWindow()
 {
 }
 
-// Method Description:
-// - Used to initialize the XAML island for the non-client area. Also calls our
-//   base IslandWindow's Initialize, which will initialize the client XAML
-//   Island.
-void NonClientIslandWindow::Initialize()
+void NonClientIslandWindow::OnDragBarSizeChanged(winrt::Windows::Foundation::IInspectable sender, winrt::Windows::UI::Xaml::SizeChangedEventArgs eventArgs)
 {
-    _nonClientSource = DesktopWindowXamlSource{};
-    auto interop = _nonClientSource.as<IDesktopWindowXamlSourceNative>();
-    winrt::check_hresult(interop->AttachToWindow(_window));
-
-    // stash the child interop handle so we can resize it when the main hwnd is resized
-    interop->get_WindowHandle(&_nonClientInteropWindowHandle);
-
-    _nonClientRootGrid = winrt::Windows::UI::Xaml::Controls::Grid{};
-
-    _nonClientSource.Content(_nonClientRootGrid);
-
-    // Call the IslandWindow Initialize to set up the client xaml island
-    IslandWindow::Initialize();
+    InvalidateRect(NULL, NULL, TRUE);
+    ForceResize();
 }
 
-// Method Description:
-// - Sets the content of the non-client area of our window to the given XAML element.
-// Arguments:
-// - content: a XAML element to use as the content of the titlebar.
-// Return Value:
-// - <none>
-void NonClientIslandWindow::SetNonClientContent(winrt::Windows::UI::Xaml::UIElement content)
+void NonClientIslandWindow::OnAppInitialized(winrt::TerminalApp::App app)
 {
-    _nonClientRootGrid.Children().Clear();
-    _nonClientRootGrid.Children().Append(content);
-}
-
+    _dragBar = app.GetDragBar();
+    _rootGrid.SizeChanged({ this, &NonClientIslandWindow::OnDragBarSizeChanged });
 
-// Method Description:
-// - Set the height we expect to reserve for the non-client content.
-// Arguments:
-// - contentHeight: the size in pixels we should use for the non-client content.
-void NonClientIslandWindow::SetNonClientHeight(const int contentHeight) noexcept
-{
-    _titlebarUnscaledContentHeight = contentHeight;
+    __super::OnAppInitialized(app);

```

Comment:
```suggestion
    IslandWindow::OnAppInitialized(app);
```
I think we'd prefer calling out the superclass specifically